### PR TITLE
feat(slashcommands): add JSON objects along with builders

### PR DESCRIPTION
**Pull request by megatank58**
**Please describe the changes this PR makes and why it should be merged:**

#1122 Adding JSON objects as an alternative option to builders will allow advanced developers to ~~copy paste~~ understand code quickly and something that is more native to them.

One question is whether the builders page should also have the JSON objects for direct comparison with builders or should they not have them as the page is specifically made for builders?

### CODE REVIEW

File: guide/interactions/slash-commands.md [hunk 1]
```diff
@@ -110,6 +113,27 @@ const data = new SlashCommandBuilder()
...
+::: code-group-item JSON
+```js
+const data = {
+	name: 'echo',
+	description: 'Replies with your input!',
+	options: [
+		{
+			name: 'input',
+			description: 'The input to echo back',
+			type: 'STRING',
```
Comment (BenjammingKirby) [inline guide/interactions/slash-commands.md:127]: just naming one out of the many, If you're actually going to use json, look at the docs, they don't take in strings, they take in numbers, the enum values
sure in v13, the applicationcommandmanager.create takes in the string i think but aren't you people discouraging this way of registering??

Comment (megatank58) [reply guide/interactions/slash-commands.md:127]: Oh i completely missed that, you are right, the raw api takes numbers

**Comment by BenjammingKirby** [comment]
I also disagree with having this in the first place, we don't need a raw json example for everything, that'd be ass to maintain and just unneeded stuff cluttering the guide imo. What I would rather do is recommend builders and link to the api docs and perhaps mention discord-api-types for typescript users. In the case of the "advanced users", that should be enough for them.

File: guide/interactions/slash-commands.md [hunk 2]
```diff
@@ -110,6 +113,28 @@ const data = new SlashCommandBuilder()
...
+
+```js
+const data = {
+	name: 'echo',
+	description: 'Replies with your input!',
+	options: [
+		{
+			name: 'input',
+			description: 'The input to echo back',
+			type: 3,
```
Comment (BenjammingKirby) [inline guide/interactions/slash-commands.md:128]: I recommended the usage of enums, is there a reason why those weren't used? They can be imported from discord-api-types
If you use rest, you need them for the routes, and rest uses it, builders uses it, so does discord.js
I don't think there's really a noticeable downside to using enums

**Review by almostSouji — CHANGES_REQUESTED** [review]
Prefer enums over magic numbers, if you really feel the need to, you can note the value as a comment.

File: guide/interactions/slash-commands.md [hunk 3]
```diff
@@ -181,6 +251,40 @@ const data = new SlashCommandBuilder()
...
+	options: [
+		{
+			name: 'user',
+			description: 'Info about a user',
+			type: ApplicationCommandOptionType.Subcommand,
+			options: [
+				{
+					name: 'target',
+					description: 'The user',
+					type: 'USER',
```
Comment (almeidx) [inline guide/interactions/slash-commands.md:272]:
```suggestion
					type: ApplicationCommandOptionType.User,
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by almostSouji** [comment]
This will need to be revisited after the semantic and structural rewrite that's currently in progress

**Comment by megatank58** [comment]
It's probably better to remake this pr entirely (or rethink the idea as a whole) as the codebase has changed a lot.

**Pull request closed without merging**